src/components/structures/RoomSearch.tsx

```diff
@@ -134,9 +136,11 @@ export default class RoomSearch extends React.PureComponent<IProps, IState> {
         }
     };
 
+
+
     private onChange = () => {
         if (this.elementRef.current?.tagName !== "INPUT") return;
-        this.setState({ query: (this.elementRef.current as HTMLInputElement).value });
+        this.setState({ query: transformSearchTerm((this.elementRef.current as HTMLInputElement).value) });
```

Comment:
I don't think this will really do anything? The `RoomSearch` just filters the rooms to already-existing ones and doesn't make suggestions to join the room you've entered, so would exclude it from this PR.